includes/sanitizers/class-amp-base-sanitizer.php

```diff
@@ -351,4 +358,25 @@ public function remove_invalid_attribute( $element, $attribute ) {
 		}
 	}
 
+	/**
+	 * Sets the current plugin that is outputting markup, if there is one.
+	 *
+	 * @since 0.7
+	 *
+	 * @param DOMNode $node The node to check for the presence of a plugin in a comment.
+	 * @return void
+	 */
+	public function set_plugin_output( $node ) {
+		if ( XML_COMMENT_NODE === $node->nodeType ) {
+			preg_match( ':(before|after)\:(.*):s', $node->nodeValue, $matches );
+			if ( ! isset( $matches[1], $matches[2] ) ) {
+				return;
+			} elseif ( 'after' === $matches[1] ) {
+				$this->current_plugin_output = $matches[2];
+			} elseif ( 'before' === $matches[1] ) {
+				$this->current_plugin_output = null;
```

Comment:
I think that `current_plugin_output` should actually be an array, a stack of sources. So this should do:

```
$this->current_source[] = $matches[2];
```

And then `above` it can just `array_pop`. This is because a theme could invoke a plugin and there can be nesting.

Reply:
Thanks, [commit](https://github.com/Automattic/amp-wp/pull/971/commits/32a58fcd21710124cf563e457880f0e80a7fb4d5) renames `$current_plugin_output` to `$current_sources`. And it makes it into an array (stack).